qiime/core/type/signature.py

```diff
@@ -9,18 +9,74 @@
 import collections
 import inspect
 import itertools
+import copy
 
 import qiime.sdk
 from .grammar import TypeExpression
 from .primitive import is_primitive_type
 from .semantic import is_semantic_type
 from .visualization import Visualization
+from ..util import ImmutableBase
+
+
+class _NoValue:
+    def __repr__(self):
+        return "NOVALUE"
+
+
+class ParameterSpec(ImmutableBase):
+    def __init__(self, **kwargs):
+        self.NOVALUE = _NoValue()
+        self.qiime_type = kwargs.pop('qiime_type', self.NOVALUE)
```

Comment:
Instead of using kwargs, is it possible to make the constructor explicitly accept these three parameters, which all have a default value of the `NOVALUE` sentinel?